internals/daemon/api_health_test.go

```diff
@@ -206,6 +208,55 @@ func (s *healthSuite) TestChecksError(c *C) {
 	})
 }
 
+// Ensure state lock is not held at all for GET /v1/health requests.
+// Regression test for issue described at:
+//
+// - https://github.com/canonical/pebble/issues/366
+// - https://bugs.launchpad.net/juju/+bug/2052517
+func (s *apiSuite) TestStateLockNotHeld(c *C) {
```

Comment:
Rather than adding the LockCount, I would have just acquired the lock, then made the call to the health endpoint. If it times out, then it depends on the lock. But at least the lock count allows you to find when you should start testing from.

Reply:
Ah, that's a very good idea! And if I do it that way it doesn't actually matter when I start testing from: whether the test acquires the lock before or after the overlord startup goroutine did or not, it doesn't matter. Fixed in https://github.com/canonical/pebble/pull/373 (and while I was in there, I found that the /v1/health error case still acquired the state lock, so I fixed that too).